superset-frontend/src/dashboard/components/Dashboard.jsx

```diff
@@ -159,11 +159,27 @@ class Dashboard extends React.PureComponent {
           // added filter?
           [].push.apply(affectedChartIds, activeFilters[filterKey].scope);
         } else {
-          // changed filter field value or scope?
-          const affectedScope = (activeFilters[filterKey].scope || []).concat(
-            appliedFilters[filterKey].scope || [],
-          );
-          [].push.apply(affectedChartIds, affectedScope);
+          // has filter field value change?
```

Comment:
NIT(optional ): `componentDidUpdate` is getting pretty large and opaque, it could be could to refactor into a `applyFilters` or some better name.

Reply:
added `applyFilters` function.